sqlglot/transforms.py

```diff
@@ -179,10 +179,12 @@ def explode_to_unnest(expression: exp.Expression) -> exp.Expression:
                 explode_alias = select.aliases[1].name
                 select = select.this
 
-            if isinstance(select, (exp.Explode, exp.Posexplode)):
+            if isinstance(select, (exp.Unnest, exp.Explode, exp.Posexplode)):
                 is_posexplode = isinstance(select, exp.Posexplode)
 
-                explode_arg = select.this
+                explode_arg = (
+                    select.expressions[0] if isinstance(select, exp.Unnest) else select.this
+                )
```

Comment:
Given that these changes were all that was required for supporting unnest here, I opted for putting it in this function. But I'd understand if you prefer putting it in a separate function